File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/queryDeviceInf/model/DeviceInfo.java
package org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf.model;

public class DeviceInfo {
    public String deviceMake;
    public String deviceModel;
    public DeviceInput deviceInput;
    public Info deviceInfo;

    class DeviceInput {
        public Integer ArrowUp;
        public Integer ArrowDown;
        public Integer ArrowRight;
        public Integer ArrowLeft;
        public Integer Select;
        public Integer Back;
    }

    class Info {
        public Integer numberOfTuners;
        public Integer yearOfMfr;
    }
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/queryDeviceInf/model/DeviceInfo.kt
package org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf.model

import org.ngbp.jsonrpc4jtestharness.rpc.IBARpc

data class DeviceInfo (
        var deviceMake: String? = null,
        var deviceModel: String? = null,
        var deviceInput: DeviceInput? = null,
        var deviceInfo: Info? = null
) : IBARpc()

data class DeviceInput (
    var ArrowUp: Int? = null,
    var ArrowDown: Int? = null,
    var ArrowRight: Int? = null,
    var ArrowLeft: Int? = null,
    var Select: Int? = null,
    var Back: Int? = null
)

data class Info (
    var numberOfTuners: Int? = null,
    var yearOfMfr: Int? = null
)